lib/ferto/client.rb

```diff
@@ -134,13 +136,19 @@ def build_header(aggr_id)
 
     def build_body(aggr_id, aggr_limit, url, callback_url, callback_type,
                    callback_dst, aggr_proxy, download_timeout, user_agent,
-                   mime_type, extra, request_headers)
+                   mime_type, extra, request_headers,
+                   s3_bucket, s3_region)
       body = {
         aggr_id: aggr_id,
         aggr_limit: aggr_limit,
         url: url
       }
 
+      if s3_bucket || s3_region
```

Comment:
Since both `s3_bucket` and `s3_region` are mandatory, I think it would be safer to use and instead of or

Reply:
Agree

Reply:
My thought was that the downloader will propagate back the error, but ok we could fail early here. I'm changing it